### PR TITLE
added [minesweeper] exercise

### PR DESCRIPTION
**Pull request by leetwinski**

### CODE REVIEW

File: exercises/minesweeper/src/example.clj [hunk 1]
```diff
@@ -0,0 +1,27 @@
+(ns minesweeper
+  (require [clojure.pprint :refer [cl-format]]
+           [clojure.string :refer [split-lines]]))
+
+(defn- windows [field]
+  (apply map vector
```
Comment (yurrriq) [inline exercises/minesweeper/src/example.clj]: Could you wrap this in `->>`?

File: exercises/minesweeper/src/example.clj [hunk 2]
```diff
@@ -0,0 +1,27 @@
+(ns minesweeper
+  (require [clojure.pprint :refer [cl-format]]
```
Comment (yurrriq) [inline exercises/minesweeper/src/example.clj]: Shouldn't this `require` be `:require`?

File: exercises/minesweeper/test/minesweeper_test.clj [hunk 3]
```diff
@@ -0,0 +1,60 @@
+(ns minesweeper-test
+  (:require [clojure.test :refer :all]
+            [clojure.string :refer [join]]
+            [minesweeper :refer :all]))
```
Comment (yurrriq) [inline exercises/minesweeper/test/minesweeper_test.clj]: I'd strongly prefer explicit imports here, i.e.

```diff
diff --git a/exercises/minesweeper/test/minesweeper_test.clj b/exercises/minesweeper/test/minesweeper_test.clj
index f39b577..d92cd76 100644
--- a/exercises/minesweeper/test/minesweeper_test.clj
+++ b/exercises/minesweeper/test/minesweeper_test.clj
@@ -1,7 +1,7 @@
 (ns minesweeper-test
-  (:require [clojure.test :refer :all]
+  (:require [clojure.test :refer [deftest is]]
             [clojure.string :refer [join]]
-            [minesweeper :refer :all]))
+            [minesweeper :refer [draw]]))
 
 (deftest zero-size-board
   (is (= (draw "") "")))
```

File: exercises/minesweeper/src/example.clj (hunk 2 above)
Comment (leetwinski) [inline exercises/minesweeper/src/example.clj]: oh! sure! Late night coding )

**Comment by leetwinski** [comment]
@yurrriq , added fixes

File: exercises/minesweeper/src/example.clj [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                            (count (filter #{\*} [a b c d f g h i])))]
+    (if (zero? mines-count)
+      \space
+      mines-count)
+    e))
+
+(defn draw [field]
+  (if (empty? field) ""
+      (->> field
+         split-lines
```
Comment (yurrriq) [inline exercises/minesweeper/src/example.clj]: This could be re-indent, but it doesn't bother me too much, I guess.

**Comment by yurrriq** [comment]
@leetwinski do you want to re-indent the other `->>` or should I just merge this now?

**Comment by leetwinski** [comment]
sure i do! just a minute

**Comment by yurrriq** [comment]
Thanks, can you pull `master` and rebase?

**Comment by leetwinski** [comment]
sure

**Comment by leetwinski** [comment]
@yurrriq , done!

**Comment by yurrriq** [comment]
Not sure what happened here, but I pulled, rebased (cleaning up the history a bit), and pushed. Thanks again!

**Comment by leetwinski** [comment]
@yurrriq , thanks for your help!

**Pull request closed without merging**